packages/maskbook/src/web3/suspends/getERC721TokenDetailed.ts

```diff
@@ -15,40 +15,30 @@ async function getERC721TokenDetailed(
     calls: any[],
     token?: Partial<ERC721TokenDetailed>,
 ) {
-    await callback(calls)
-    // compose
-    const [name, symbol, baseURI] = results.map((x) => (x.error ? undefined : x.value))
-    const token_ = {
-        type: EthereumTokenType.ERC721,
-        address: formatChecksumAddress(address),
-        chainId,
-        name: name ?? token?.name ?? '',
-        symbol: symbol ?? token?.symbol ?? '',
-        baseURI: baseURI ?? token?.baseURI ?? '',
-    } as ERC721TokenDetailed
-
-    if (!cache.has(address)) {
-        try {
-            cache.set(address, token_)
-        } catch (error) {
-            throw error
-        }
+    try {
+        await callback(calls)
+        // compose
+        const [name, symbol, baseURI] = results.map((x) => (x.error ? undefined : x.value))
+        const token_ = {
+            type: EthereumTokenType.ERC721,
+            address: formatChecksumAddress(address),
+            chainId,
+            name: name ?? token?.name ?? '',
+            symbol: symbol ?? token?.symbol ?? '',
+            baseURI: baseURI ?? token?.baseURI ?? '',
+        } as ERC721TokenDetailed
+        cache.set(address, token_)
+    } catch {
+        cache.set(address, {} as ERC721TokenDetailed)
```

Comment:
@lelenei wait, this is not a the best way to cause an error.

Reply:
so why not throw error directly?

Reply:
As I mentioned here: https://git.io/JtWyf